From 44340aa491d7c6601f621eb3093fd1ce7e30c048 Mon Sep 17 00:00:00 2001
From: OliverG96 <oliverarafo@gmail.com>
Date: Sun, 9 Dec 2012 12:21:08 +0000
Subject: [PATCH] display: Fix build error related to vSync

---
 libhwcomposer/hwc_vsync.cpp |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/libhwcomposer/hwc_vsync.cpp b/libhwcomposer/hwc_vsync.cpp
index a3becfa..c66ab5c 100644
--- a/libhwcomposer/hwc_vsync.cpp
+++ b/libhwcomposer/hwc_vsync.cpp
@@ -31,8 +31,8 @@
 #include "string.h"
 #include "external.h"
 
-
 namespace qhwc {
+#ifndef NO_HW_VSYNC
 
 #define HWC_VSYNC_THREAD_NAME "hwcVsyncThread"
 
@@ -87,7 +87,6 @@ static void *vsync_loop(void *param)
             pthread_cond_wait(&ctx->vstate.cond, &ctx->vstate.lock);
         }
         pthread_mutex_unlock(&ctx->vstate.lock);
-
         if (!enabled) {
             int e = 1;
             if(ioctl(ctx->dpyAttr[dpy].fd, MSMFB_OVERLAY_VSYNC_CTRL,
@@ -149,5 +148,5 @@ void init_vsync_thread(hwc_context_t* ctx)
             HWC_VSYNC_THREAD_NAME, strerror(ret));
     }
 }
-
+#endif
 }; //namespace
-- 
1.7.10.4

