From ed1bd077c45f3986c7888763c4a8b6e7d20de147 Mon Sep 17 00:00:00 2001
From: WinSuk <goodboy735@gmail.com>
Date: Sat, 17 Nov 2012 16:46:53 -0500
Subject: [PATCH 3/6] Fix the MSM7x27 encoder

---
 media/libstagefright/OMXCodec.cpp |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/media/libstagefright/OMXCodec.cpp b/media/libstagefright/OMXCodec.cpp
index d9df4ca..3b8581e 100644
--- a/media/libstagefright/OMXCodec.cpp
+++ b/media/libstagefright/OMXCodec.cpp
@@ -808,6 +808,10 @@ status_t OMXCodec::configureCodec(const sp<MetaData> &meta) {
             status_t err = mOMX->getExtensionIndex(
                 mNode, OMX_QCOM_INDEX_PARAM_VIDEO_SYNCFRAMEDECODINGMODE, &indexType);
 
+#ifdef QCOM_LEGACY_OMX
+            // Don't run this check with the legacy encoder
+            if (strncmp(mComponentName, "OMX.qcom.video.encoder.", 23))
+#endif
             CHECK_EQ(err, (status_t)OK);
 
             enableType.bEnable = OMX_TRUE;
-- 
1.7.10.4

