From 3b17871a080afc4084bd724df830205391c4411a Mon Sep 17 00:00:00 2001
From: Ganster41 <ganster0x29@gmail.com>
Date: Thu, 22 Nov 2012 10:58:36 +0200
Subject: [PATCH 5/6] Fix using extended extractor from older Androids

---
 media/libstagefright/DataSource.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/media/libstagefright/DataSource.cpp b/media/libstagefright/DataSource.cpp
index 8e61719..9d66486 100644
--- a/media/libstagefright/DataSource.cpp
+++ b/media/libstagefright/DataSource.cpp
@@ -95,7 +95,7 @@ bool DataSource::sniff(
     for (List<SnifferFunc>::iterator it = gSniffers.begin();
          it != gSniffers.end(); ++it) {
 
-#ifdef QCOM_HARDWARE
+#if defined(QCOM_HARDWARE) && !defined(QCOM_LEGACY_OMX)
         //Dont call the first sniffer from extended extarctor
         if(it == extendedSnifferPosition)
             continue;
-- 
1.7.10.4

